src/simplnx/DataStructure/DataStore.hpp

```diff
@@ -82,6 +83,47 @@ class DataStore : public AbstractDataStore<T>
   , m_NumComponents(std::accumulate(m_ComponentShape.cbegin(), m_ComponentShape.cend(), static_cast<size_t>(1), std::multiplies<>()))
   , m_NumTuples(std::accumulate(m_TupleShape.cbegin(), m_TupleShape.cend(), static_cast<size_t>(1), std::multiplies<>()))
   {
+    // Because no init value is passed into the constructor, we will use a "mudflap" style value that is easy to debug.
+    if constexpr(sizeof(T) == 1)
```

Comment:
```cpp
template <class T>
constexpr T GetMudflap() noexcept
{
  if constexpr(sizeof(T) == 1)
  {
    return std::bit_cast<T>(static_cast<uint8>(0xAB));
  }
  if constexpr(sizeof(T) == 2)
  {
    return std::bit_cast<T>(static_cast<uint16>(0xABAB));
  }
  if constexpr(sizeof(T) == 4)
  {
    return std::bit_cast<T>(static_cast<uint32>(0xABABABAB));
  }
  if constexpr(sizeof(T) == 8)
  {
    return std::bit_cast<T>(static_cast<uint64>(0xABABABABABABABAB));
  }
}
```
Something like this would work.